Physicc/src/bvh.cpp

```diff
@@ -1,42 +1,143 @@
 #include "tools/Tracy.hpp"
+/**
+ * @file bvh.cpp
+ * @brief Constructs a BVH given a list of RigidBody objects.
+ * @author Divyansh Tiwari (divyanshtiwari237)
+ * @author Tirthankar mazumder (wermos)
+ * @bug No known bugs.
+ */
+
+/* -- Includes -- */
+/*bvh header*/
 
 #include "bvh.hpp"
 
 #include <utility>
+#include <algorithm>
 
 namespace Physicc
 {
-	using Iterator = std::vector<RigidBody>::iterator;
-
 	BVH::BVH(std::vector<RigidBody> rigidBodyList)
 		: 	m_head(nullptr),
 			m_rigidBodyList(std::move(rigidBodyList)) 
 	{
 	}
 
-	inline void BVH::buildTree()
+	BoundingVolume::AABB BVH::computeBV(int start, int end)
```

Comment:
These function signatures (this one, the line on line 41, the one line 107, and the one on line 140) will also need to be changed to mirror the changes in the corresponding header file.